### ISSUE TITLE
[BUG] ExecutionException was thrown when using DefaultAzureCredential to get accessToken with azure-identity:1.6.0-beta.1

### ISSUE THREAD

**Issue by backwind1233** [Azure/azure-sdk-for-java#30983]
**Describe the bug**
Use DefaultAzureCredential to get accessToken but throw Exception.

***Exception or Stack Trace***
```
21:57:33.891 [ForkJoinPool.commonPool-worker-1] ERROR com.microsoft.aad.msal4j.ConfidentialClientApplication - [Correlation ID: 15810490-0d39-482e-8eb5-82227fefaa6a] Execution of class com.microsoft.aad.msal4j.AcquireTokenByClientCredentialSupplier failed.
java.util.concurrent.ExecutionException: com.azure.identity.CredentialUnavailableException: ManagedIdentityCredential authentication unavailable. Connection to IMDS endpoint cannot be established, Connect timed out.
	at java.base/java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:396)
	at java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2073)
	at com.microsoft.aad.msal4j.AcquireTokenByAppProviderSupplier.fetchTokenUsingAppTokenProvider(AcquireTokenByAppProviderSupplier.java:61)
	at com.microsoft.aad.msal4j.AcquireTokenByAppProviderSupplier.execute(AcquireTokenByAppProviderSupplier.java:44)
	at com.microsoft.aad.msal4j.AcquireTokenByClientCredentialSupplier.acquireTokenByClientCredential(AcquireTokenByClientCredentialSupplier.java:78)
	at com.microsoft.aad.msal4j.AcquireTokenByClientCredentialSupplier.execute(AcquireTokenByClientCredentialSupplier.java:49)
	at com.microsoft.aad.msal4j.AuthenticationResultSupplier.get(AuthenticationResultSupplier.java:71)
	at com.microsoft.aad.msal4j.AuthenticationResultSupplier.get(AuthenticationResultSupplier.java:20)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1768)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.exec(CompletableFuture.java:1760)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)
Caused by: com.azure.identity.CredentialUnavailableException: ManagedIdentityCredential authentication unavailable. Connection to IMDS endpoint cannot be established, Connect timed out.
	at com.azure.identity.implementation.IdentityClient.lambda$checkIMDSAvailable$65(IdentityClient.java:1574)
	at reactor.core.publisher.MonoCallable.call(MonoCallable.java:92)
	at reactor.core.publisher.FluxFlatMap.trySubscribeScalarMap(FluxFlatMap.java:127)
	at reactor.core.publisher.MonoFlatMap.subscribeOrReturn(MonoFlatMap.java:53)
	at reactor.core.publisher.Mono.subscribe(Mono.java:4382)
	at reactor.core.publisher.Mono.subscribeWith(Mono.java:4512)
	at reactor.core.publisher.Mono.toFuture(Mono.java:4917)
	at com.azure.identity.implementation.IdentityClient.lambda$getManagedIdentityConfidentialClient$8(IdentityClient.java:320)
	at com.microsoft.aad.msal4j.AcquireTokenByAppProviderSupplier.fetchTokenUsingAppTokenProvider(AcquireTokenByAppProviderSupplier.java:59)
	... 12 common frames omitted
Caused by: java.net.SocketTimeoutException: Connect timed out
	at java.base/sun.nio.ch.NioSocketImpl.timedFinishConnect(NioSocketImpl.java:543)
	at java.base/sun.nio.ch.NioSocketImpl.connect(NioSocketImpl.java:594)
	at java.base/java.net.Socket.connect(Socket.java:633)
	at java.base/sun.net.NetworkClient.doConnect(NetworkClient.java:178)
	at java.base/sun.net.www.http.HttpClient.openServer(HttpClient.java:498)
	at java.base/sun.net.www.http.HttpClient.openServer(HttpClient.java:603)
	at java.base/sun.net.www.http.HttpClient.<init>(HttpClient.java:246)
	at java.base/sun.net.www.http.HttpClient.New(HttpClient.java:351)
	at java.base/sun.net.www.http.HttpClient.New(HttpClient.java:373)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getNewHttpClient(HttpURLConnection.java:1309)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.plainConnect0(HttpURLConnection.java:1242)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.plainConnect(HttpURLConnection.java:1128)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.connect(HttpURLConnection.java:1057)
	at com.azure.identity.implementation.IdentityClient.lambda$checkIMDSAvailable$65(IdentityClient.java:1568)
	... 20 common frames omitted
```

**To Reproduce**
1. add dependence
```xml
<dependency>
	<groupId>com.azure</groupId>
	<artifactId>azure-identity</artifactId>
	<version>1.6.0-beta.1</version>
</dependency>
```

2. Login with Azure CLI
3. Add codes
```java
   public static void main(String[] args) {
        DefaultAzureCredential dac = new DefaultAzureCredentialBuilder().build();
        TokenRequestContext requestContext = new TokenRequestContext();
        requestContext.addScopes("https://ossrdbms-aad.database.windows.net/.default");
        String token = dac.getToken(requestContext).block().getToken();
        System.out.println("token = " + token);
    }
```
4. Run the code.

***Code Snippet***
Add the code snippet that causes the issue.

**Expected behavior**
I hope it will return accessToken successfully.

**Screenshots**

**Setup (please complete the following information):**
 - OS:  MAC OS
 - IDE: [e.g. IntelliJ] IntelliJ
 - Library/Libraries:  azure-identity:1.6.0-beta.1
 - Java version: Java 18
 - App Server/Environment: NA
 - Frameworks: NA

If you suspect a dependency version mismatch (e.g. you see `NoClassDefFoundError`, `NoSuchMethodError` or similar), please check out [Troubleshoot dependency version conflict article](https://aka.ms/azsdk/java/dependency/troubleshoot) first. If it doesn't provide solution for the problem, please provide:
- verbose dependency tree (`mvn dependency:tree -Dverbose`)
- exception message, full stack trace, and any available logs

**Additional context**

**Information Checklist**
Kindly make sure that you have added all the following information above and checkoff the required fields otherwise we will treat the issuer as an incomplete report
- [x] Bug Description Added
- [ ] Repro Steps Added
- [ ] Setup information Added

**Comment by backwind1233** [comment]
Hi @g2vinay, let's use this to track.

**Comment by joshfree** [comment]
cc: @billwert

**Comment by stliu** [comment]
@g2vinay @joshfree Hi we're going to release our jdbc auth plugin next week and this is a blocking issue, please help to get this fixed and released asap, thanks.

**Comment by billwert** [comment]
Hello! Do you need to use the beta library? Does this reproduce with 1.5.5?

**Comment by g2vinay** [comment]
Resolved in v1.7.0-beta.1
